app/mockServices/mockData.service.ts

```diff
@@ -19,30 +28,188 @@ class Data {
 }
 
 export class DataService implements IDataService {
-  public static $inject = ['$q'];
+  public static $inject = ['$q', 'APIService'];
 
   private $q : any;
+  private APIService: any;
+  private watchCallbacksMap: any = {};
 
-  constructor ($q: angular.IQService) {
+
+  constructor ($q: angular.IQService, APIService: any) {
     this.$q = $q;
+    this.APIService = APIService;
   }
 
-  public list (resource: any, context: any, callback: any, opts: any) : angular.IPromise < any > {
+  public list(resource: any, context: any, callback: any, opts: any) : angular.IPromise < any > {
     let deferred = this.$q.defer();
     if (callback) {
       deferred.promise.then(callback);
     }
 
-    if (resource.resource === 'serviceclasses') {
-      setTimeout(() => {
-        deferred.resolve(new Data(angular.copy(servicesData)));
-      }, 300);
-    } else if (resource === 'imagestreams' && context.namespace === 'openshift') {
-      setTimeout(() => {
-        deferred.resolve(new Data(angular.copy(imagesData)));
-      }, 300);
+    var data: any = this.getMockData(resource, context);
+
+    deferred.resolve(data);
+
+    return deferred.promise;
+  }
+
+  public delete(resource: any, name: any, context: any, opts: any) : angular.IPromise < any > {
+    let deferred = this.$q.defer();
+    var data: any = this.getMockData(resource, context);
+
+    _.remove(data.by(), function(nextData: any) {
+      return nextData.metadata.name === name;
+    });
+    setTimeout(() => {
```

Comment:
Should this be `$timeout` so deferred is resolved in a digest loop? Here and below.